From 059b41325f68cc37a79d602ff271873487ba02ba Mon Sep 17 00:00:00 2001
From: Siew Chin Lim <elly.siew.chin.lim@intel.com>
Date: Sun, 5 Sep 2021 02:52:18 +0800
Subject: [PATCH 01/20] HSD #22013762153: intel: fcs: Increase size of struct
 intel_fcs_dev_ioctl

commit 53dac822e8c6a1c580f05a9e96dfb818d76fac34 from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Increase size of struct intel_fcs_dev_ioctl to 256 bytes to leave more
empty space for future expansion. The purpose is to prevent size from
changing when adding new feature. Size change will cause incompatibility
issue between Linux FCS driver and user space application.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 include/uapi/linux/intel_fcs-ioctl.h | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/include/uapi/linux/intel_fcs-ioctl.h b/include/uapi/linux/intel_fcs-ioctl.h
index c190f5de5009..1316a24490f1 100644
--- a/include/uapi/linux/intel_fcs-ioctl.h
+++ b/include/uapi/linux/intel_fcs-ioctl.h
@@ -11,6 +11,13 @@
 /* the value may need be changed when upstream */
 #define INTEL_FCS_IOCTL		0xC0
 
+/* define macro to be used to fix the size of struct intel_fcs_dev_ioctl */
+#define INTEL_FCS_IOCTL_MAX_SZ		256U
+/* the header include the 8 bytes stucture padding and 4 bytes status */
+#define INTEL_FCS_IOCTL_HEADER_SZ	12U
+#define INTEL_FCS_IOCTL_PLACEHOLDER_SZ	(INTEL_FCS_IOCTL_MAX_SZ - \
+					 INTEL_FCS_IOCTL_HEADER_SZ) / 4
+
 /**
  * enum fcs_vab_img_type - enumeration of image types
  * @INTEL_FCS_IMAGE_HPS: Image to validate is HPS image
@@ -31,6 +38,14 @@ enum fcs_certificate_test {
 	INTEL_FCS_TEST = 1
 };
 
+/**
+ * struct fcs_placeholder - placeholder of ioctl stuct
+ * @data: placeholder of iotcl struct
+ */
+struct fcs_placeholder {
+	uint32_t data[INTEL_FCS_IOCTL_PLACEHOLDER_SZ];
+};
+
 /**
  * struct intel_fcs_cert_test_word - certificate test word
  * @test_word: if set, do not write fuses, write to cache only.
@@ -390,7 +405,7 @@ struct fcs_sdos_data_ext {
 };
 
 /**
- * struct intel_fcs_dev_ioct: common structure passed to Linux
+ * struct intel_fcs_dev_ioctl: common structure passed to Linux
  *	kernel driver for all commands.
  * @status: Used for the return code.
  *      -1 -- operation is not started
@@ -412,6 +427,7 @@ struct intel_fcs_dev_ioctl {
 
 	/* command parameters */
 	union {
+		struct fcs_placeholder		placeholder;
 		struct fcs_validation_request	s_request;
 		struct fcs_certificate_request	c_request;
 		struct fcs_single_certificate_request	i_request;
-- 
2.31.1

